pkg/plugins/manager/loader/assetpath/assetpath.go

```diff
@@ -16,43 +17,39 @@ import (
 // on the plugins CDN, and it will switch to the correct implementation depending on the plugin and the config.
 type Service struct {
 	cdn *pluginscdn.Service
+	cfg *config.Cfg
 }
 
-func ProvideService(cdn *pluginscdn.Service) *Service {
-	return &Service{cdn: cdn}
+func ProvideService(cfg *config.Cfg, cdn *pluginscdn.Service) *Service {
+	return &Service{cdn: cdn, cfg: cfg}
 }
 
 // Base returns the base path for the specified plugin.
 func (s *Service) Base(pluginJSON plugins.JSONData, class plugins.Class, pluginDir string) (string, error) {
 	if class == plugins.ClassCore {
-		return path.Join("public/app/plugins", string(pluginJSON.Type), filepath.Base(pluginDir)), nil
+		return path.Join("/", s.cfg.GrafanaAppSubURL, "/public/app/plugins", string(pluginJSON.Type), filepath.Base(pluginDir)), nil
 	}
 	if s.cdn.PluginSupported(pluginJSON.ID) {
-		return s.cdn.SystemJSAssetPath(pluginJSON.ID, pluginJSON.Info.Version, "")
+		return s.cdn.AssetURL(pluginJSON.ID, pluginJSON.Info.Version, "")
 	}
-	return path.Join("public/plugins", pluginJSON.ID), nil
+	return path.Join("/", s.cfg.GrafanaAppSubURL, "/public/plugins", pluginJSON.ID), nil
 }
 
 // Module returns the module.js path for the specified plugin.
 func (s *Service) Module(pluginJSON plugins.JSONData, class plugins.Class, pluginDir string) (string, error) {
 	if class == plugins.ClassCore {
-		return path.Join("app/plugins", string(pluginJSON.Type), filepath.Base(pluginDir), "module"), nil
+		return path.Join("core:plugin", filepath.Base(pluginDir)), nil
 	}
 	if s.cdn.PluginSupported(pluginJSON.ID) {
-		return s.cdn.SystemJSAssetPath(pluginJSON.ID, pluginJSON.Info.Version, "module")
+		return s.cdn.AssetURL(pluginJSON.ID, pluginJSON.Info.Version, "module.js")
 	}
-	return path.Join("plugins", pluginJSON.ID, "module"), nil
+	return path.Join("/", s.cfg.GrafanaAppSubURL, "/public/plugins", pluginJSON.ID, "module.js"), nil
 }
 
 // RelativeURL returns the relative URL for an arbitrary plugin asset.
-// If pathStr is an empty string, defaultStr is returned.
-func (s *Service) RelativeURL(p *plugins.Plugin, pathStr, defaultStr string) (string, error) {
-	if pathStr == "" {
-		return defaultStr, nil
-	}
-	if s.cdn.PluginSupported(p.ID) {
-		// CDN
-		return s.cdn.NewCDNURLConstructor(p.ID, p.Info.Version).StringPath(pathStr)
+func (s *Service) RelativeURL(pluginJSON plugins.JSONData, class plugins.Class, pluginDir, pathStr string) (string, error) {
```

Comment:
Yeah it's a fair point. The main thing I didn't like was relying on the `BaseURL` field already being set prior to calling the function, so I can take a look at tidying up further. One other thing that irks me is that because plugin.Plugin is being actively _built_ throughout all the loader steps, it feels a bit weird to know what exactly you can rely on from within the function. Maybe another mini struct would help 🤔 May be overthinking, but will fiddle some more

Reply:
Ah yeah, I hadn't thought about the fact that it's being constructed. Maybe we can just add a docstring to make it clearer how the function works and what the args are. Something like:

```go
// RelativeURL returns the relative URL for an arbitrary plugin asset, which can be used by the browser to load plugin assets.
//
// If the plugin ID is supported by the plugins CDN, it returns an absolute plugins CDN URL to the specified asset.
// Otherwise, it returns a relative URL that points to the asset served by the Grafana instance.
//
// pluginDir is the plugin's base dir, which is used when serving assets without the plugins CDN.
//
// pathStr is the relative path to the plugin asset.
```

Reply:
Have since updated the code - LMK is that's better and we can go from there